### PR TITLE
Feature/endpoint form

### PR DESCRIPTION
**Pull request by Cich0sza**

### CODE REVIEW

File: cogboard-webapp/src/actions/thunks.js [hunk 1]
```diff
@@ -126,6 +128,64 @@ const reorderWidgetsThunk = (sourceId, targetIndex) => (dispatch, getState) => {
   dispatch(sortWidgets({ sourceId, targetIndex, boardId }));
 };
 
+const loadSettingsThunk = token => dispatch => {
+  const init = {
+    headers: {
+      Authorization: token ? token : getToken()
```
Comment (tMaxx) [inline cogboard-webapp/src/actions/thunks.js]: `token || getToken()`?

Comment (Cich0sza) [reply cogboard-webapp/src/actions/thunks.js]: Ups. It was for test purpose, I should remove it. Fixed. ;)

File: cogboard-webapp/src/components/AddCredential.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  const handleSubmit = values => {
+    delete values.passwordConfirmation;
+
+    dispatch(saveCredential(values));
+    handleDialogClose();
+  };
+
+  return (
+    <>
+      {largeButton ? (
```
Comment (tMaxx) [inline cogboard-webapp/src/components/AddCredential.js]: Could you extract this ternary statement to a separate 'something'? Call me old fashioned, but I feel this is a case for a normal `if`.

File: cogboard-webapp/src/components/AddEndpoint.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    dispatch(saveEndpoint(values));
+    if (dataChanged !== undefined) {
+      dataChanged();
+    }
+    handleDialogClose();
+  };
+
+  return (
+    <>
+      {largeButton ? (
```
Comment (tMaxx) [inline cogboard-webapp/src/components/AddEndpoint.js]: Same thing here, you even duplicate the code. Maybe extract to a separate component?

File: cogboard-webapp/src/components/EditCredential.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      fetch(`${URL.CREDENTIALS_ENDPOINT}/${id}`, init)
+        .then(response => response.json())
+        .then(data => setCredentialData(data))
+        .catch(console.error);
+    }
+  }, [isAuthenticated, id, dialogOpened]);
+
+  return (
+    <>
+      <IconButton
```
Comment (tMaxx) [inline cogboard-webapp/src/components/EditCredential.js]: I see a lot of similarly-looking sequences of `IconButton` followed by `AppDialog`. Maybe we need another super-component for this usecase?

File: cogboard-webapp/src/helpers/index.js [hunk 5]
```diff
@@ -22,7 +22,6 @@ export const parseYupErrors = errors => {
   let result = {};
   errors.inner.forEach(error => {
     const { path, message } = error;
-
```
Comment (tMaxx) [inline cogboard-webapp/src/helpers/index.js]: Leave this whitespace alone, it didn't kill anyone 😛

File: knotx/conf/openapi.yaml [hunk 6]
```diff
@@ -13,8 +13,8 @@ servers:
         description: api domain
       port:
         enum:
-          - '8092'
-        default: '8092'
+          - "8092"
```
Comment (tMaxx) [inline knotx/conf/openapi.yaml]: I'd advise to turn off this kind of linter if you have it on, you shouldn't even touch this file for the scope of this task.

Comment (Cich0sza) [reply knotx/conf/openapi.yaml]: This is probably the work of our Prittier which works when committing, I have nothing to do with it: D

File: cogboard-webapp/src/reducers/app/settings.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  credentials: []
+};
+
+const settings = (state = initState, { type, payload }) => {
+  switch (type) {
+    case SAVE_SETTINGS:
+      return {
+        ...state,
+        ...payload
+      };
```
Comment (goeson00) [inline cogboard-webapp/src/reducers/app/settings.js]: We have now ramda library so please use assoc function to update object.

File: knotx/conf/openapi.yaml (hunk 6 above)
Comment (szymon-owczarzak) [reply knotx/conf/openapi.yaml]: **if this is prettier issue than we have a problem - it shouldn't touch anything but webapp sources**

File: cogboard-webapp/src/components/EditEndpoint.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+            id={id}
+            {...endpointData}
+          />
+        )}
+      </AppDialog>
+    </>
+  );
+};
+
+export default EditEndpoint;
```
Comment (goeson00) [inline cogboard-webapp/src/components/EditEndpoint.js]: We need to investigate further on more generic solution like using HOC, render prop or component as a prop.

File: cogboard-webapp/src/components/MainTemplate/styled.js [hunk 9]
```diff
@@ -50,3 +52,15 @@ export const StyledDrawer = styled(props => (
     background: ${getBackgroundColor('default')};
   }
 `;
+
+export const drawerStyles = makeStyles({
+  root: {
+    justifyContent: 'space-between'
+  }
+});
```
Comment (goeson00) [inline cogboard-webapp/src/components/MainTemplate/styled.js]: You can access the root using styled like this `styled(props => <Component classes={{root: 'root'}} {...props} />)` root: styles... ``

**Pull request merged**